pandas/core/indexing.py

```diff
@@ -1408,7 +1409,7 @@ def __getitem__(self, key):
             maybe_callable = com.apply_if_callable(key, self.obj)
             return self._getitem_axis(maybe_callable, axis=axis)
 
-    def _is_scalar_access(self, key):
+    def _is_scalar_access(self, key: Tuple):
```

Comment:
Length here is always indeterminate right? If it is predefined would be nice to subscript with the length of elements

Reply:
Yah, in general we don't know the length.

I'll be doing some more work in this file, will try to make types more specific as I figure them out